enclave/core/sgx/tracee.c

```diff
@@ -8,25 +8,24 @@
 
 #include "../tracee.h"
 #include "report.h"
+#include "tracee.h"
 
-static volatile int _is_enclave_debug_allowed = -1;
+static volatile int _is_debug_enclave = -1;
 
-// Read an enclave's identity attribute to see to if it was signed as an debug
-// enclave
-bool is_enclave_debug_allowed()
+static bool _is_enclave_debug_allowed(oe_sgx_td_t* td)
 {
-    oe_sgx_td_t* td = NULL;
-    int debug_allowed =
-        __atomic_load_n(&_is_enclave_debug_allowed, __ATOMIC_ACQUIRE);
+    oe_sgx_td_t* td_local = NULL;
+    int debug_allowed = __atomic_load_n(&_is_debug_enclave, __ATOMIC_ACQUIRE);
 
     if (debug_allowed != -1)
         goto done;
 
     // Start off by assuming debug is not allowed.
     debug_allowed = 0;
-    td = oe_sgx_get_td();
 
-    if (td && td->simulate)
+    td_local = td ? td : oe_sgx_get_td();
```

Comment:
Since only the `simulate` field from td is needed, can we instead expose that via a `oe_sgx_is_simulation_mode_enclave()`. That function could get a possibly uninitialized td and fetch the appropriate field.